### DISCUSSION TITLE
testFundFailWithoutEnoughETH

### DISCUSSION THREAD

**Discussion by ProfSaz** [Cyfrin/foundry-full-course-cu#1487] [Q&A] [answered]
Test passes when there is no revert and fails when there is revert which is supposed to be the other way round 

 //The code 
 function testFundFailWithoutEnoughETH() public {
        vm.expectRevert();
        fundMe.fund();
    }

//The failed log

  [20328] FundMeTest::testFailedWhenNotEnoughEthSent()
    ├─ [14883] FundMe::fund()
    │   ├─ [8993] MockV3Aggregator::latestRoundData() [staticcall]
    │   │   └─ ← 1, 200000000000 [2e11], 1, 1, 1
    │   └─ ← revert: You need to spend more ETH!
    └─ ← ()

**Comment by EngrPips** [comment]
Hello @ProfSaz , please run `foundryUp` and then retry the test. let me know if that works

**Comment by fabriziogianni7** [comment, marked as answer]
looking at VM  contract you have these 3 signatures:

![Schermata 2024-03-25 alle 09 44 22](https://github.com/Cyfrin/foundry-full-course-f23/assets/46995085/0e3c99f0-4aad-4a73-9832-f00fdde90a0d)

debugging I figured out that this line's ` require(msg.value.getConversionRate(s_priceFeed) >= MINIMUM_USD, "You need to spend more ETH!");` error message always start with "revert"
so if in the test you put `vm.expectRevert("revert: You need to spend more ETH!");` it will work.

![Schermata 2024-03-25 alle 09 46 57](https://github.com/Cyfrin/foundry-full-course-f23/assets/46995085/00a61825-3367-46de-90ca-bb323e21fd1f)

I just copied the error message I got after running the test with -vvv
![Schermata 2024-03-25 alle 09 47 18](https://github.com/Cyfrin/foundry-full-course-f23/assets/46995085/513a39b9-53a2-42e9-ac44-fe473f90e093)

It will also work if you just put the beginning of the error message, like for instance "revert"

**Comment by EngrPips** [comment]
Hello @ProfSaz ,  @fabriziogianni7  have helped with an answer please let us know if it fixes your issue.

**Comment by ProfSaz** [comment]
Yeah it is solved thank you @fabriziogianni7 and @EngrPips
